doc/decisions/private_api.md

```diff
@@ -0,0 +1,50 @@
+# Private API
+
+## Problem
+
+Only `libelektra-core` is supposed to access private data, but this contradicts the goal to keep the library minimal.
+`kdbprivate.h` was too generic, it contained many other parts next to the struct definitions of Key/KeySet.
+
+## Constraints
+
+## Assumptions
+
+- The [C99 standard, section 5.2.4.1](http://www.open-std.org/jtc1/sc22/wg14/) gives following limit:
+  4095 external identifiers in one translation unit
+
+## Considered Alternatives
+
+- keep current situation
+
+## Decision
+
+- Also allow other libraries (e.g. a new `libelektra-operations`) to access to non-public API.
+- Put struct definitions of Key/KeySet in a separate header file, which gets included by parts that need it (see also [Header File Structure](header_file_structure.md)).
+
+- Plugins can define compatible ranges of core API version via the contract. This version is checked during `elektraPluginOpen` by `libelektra-kdb`.
```

Comment:
Is this already implemented in new-backend. We probably need an issue to track this.

Reply:
No it is not implemented. But we removed the extern consts for the version yesterday, so I'd just remove this from the decision too.

```suggestion
```